plugins/SitesManager/SitesManager.php

```diff
@@ -68,10 +69,30 @@ public function redirectDashboardToWelcomePage(&$module, &$action)
         // Skip the screen if purging logs is enabled
         $settings = PrivacyManager::getPurgeDataSettings();
         if ($settings['delete_logs_enable'] == 1) {
+            $hadTrafficKey = 'SitesManagerHadTrafficInPast_' . (int) $siteId;
+            $hadTrafficBefore = Option::get($hadTrafficKey);
+            if (!empty($hadTrafficBefore)) {
+                // user had traffic at some stage in the past... not needed to show tracking code
+                return;
+            } elseif (self::hasTrackedAnyTraffic($siteId)) {
+                // remember the user had traffic in the past so we won't show the tracking screen again
+                // if all visits are deleted for example
+                Option::set($hadTrafficKey, 1);
+                return;
+            } else {
+                // never had any traffic
+                $session = new SessionNamespace('siteWithoutData');
+                if (!empty($session->ignoreMessage)) {
+                    return;
+                }
+
+                $module = 'SitesManager';
+                $action = 'siteWithoutData';
+            }
             return;
         }
 
-        if (self::hasTrackedAnyTraffic($siteId)) {
+        if (!self::hasTrackedAnyTraffic($siteId)) {
```

Comment:
`delete_logs_enable` can be turned on/off, correct? Should we set the above option here too?

Reply:
in an 'else' I mean

Reply:
:+1: pushing it now. That should make it a lot simpler